website/content/docs/overview/modules/index.md

```diff
@@ -6,120 +6,60 @@ weight: 40
 ---
 
 The project currently has two extension mechanisms, namely connectors and modules. 
-Here we describe what modules are and what default modules are installed using the default {{< name >}} installation.
+This page describes what modules are and how they are leveraged by the control plane to build the data plane flow.  
 
-# What are modules?
+As described in the [Architecture]({{< baseurl >}}/docs/overview/architecture/) page, the control plane generates a description of a data plane based on policies and application requirements. This is known as a blueprint, and includes components that are deployed by the control plane to fulfill different data-centric requirements.  For example, a component that can mask data can be used to enforce a data masking policy, or a component that copies data may be used to create a local data copy to meet performace requirements, etc. Modules are the way to describe such data plane components and make them available to the control plane. 
 
-At runtime, the control plane builds a blueprint for each application. The blueprint includes additional components that are injected into the data path by the control plane to fulfill governance requirements and application requirements.
+The functionality described by the module may be deployed (a) per workload, or (b) it may be a component that runs independent of the workload and its associated control plane.  In the case of (a), the Mesh for Data control plane handles the deployment of the functional component. In the case of (b) where the functionality of the module runs independently and handles requests from multiple workloads, a client module is what is deployed by the Mesh for Data control plane.  This client module passes parameters to the external component and monitors that status and results of the request to the external component.  The external component is declared as a dependency in the module yaml.
```

Comment:
I find this description too complex. It has a subjective view on what "functionality" means (I assume you meant the direct read/write functionality here) and it targets the specific case of single external dependency that you delegate all of the "functionality" to. That's not always the case.

Instead, you can think of this as: a module can use external services (one or more) that run outside of the scope of M4D. These must be explicitly declared in the M4DModule as dependencies.

Reply:
@roee88 Updated to address the possibility of multiple external components.  As for the meaning of functionality, I don't think it will be limited to read/write.  Some examples: DataQuality for example may have a server that does quality calculations.  Or an SLA module might provide metrics regarding the contract vs. actual SLA.

Reply:
I don't see an update, did you push? 

Anyway, it's not just the case of having more than one external component. I talk about viewing it as a "client" to something that implements the logic. That's an implementation detail of a specific case. I can create a module that uses an external component to do a tiny portion of the logic and what you call "client" will do the actual logic. The "client" based description is misleading.

Reply:
@roee88 I agree that the module could have only a small part of its logic implemented externally, but it is still a client of those components so I'm not sure why this is a problem.  Based on the people/groups I am talking to who want to contribute modules I think this description will be the most understandable to them.  If not, we can always update it in a future PR.  I have meetings today and tomorrow around this, so I hope we can merge and continue the discussion ... assuming this isn't a major issue in your opinion.